### ISSUE TITLE
Short circuit cait-sith protocols on error

### ISSUE THREAD

**Issue by volovyks** [sig-net/mpc#34]

**Comment by jakmeier** [comment]
Not much info in this issue :) But the title caught my interest.

What is the status here?

I guess short-circuiting would mean something like a broadcast to all nodes that they can forget a currently running protocol instance. Given the security assumptions of cait-sith, I believe we would also need to ensure the inputs used are properly deleted. So probably we would need a robust way aborting a protocol that's a bit more reliable than fire-and-forget.

**Comment by volovyks** [comment]
Yes, the idea is to stop the protocol and not waste time and resources if one or more nodes know it failed. The most popular case is when one of the nodes does not have the requested triple or presignature.
It also affects signature protocol latency, as we are now waiting for a protocol to finish, but it never will.

**Comment by volovyks** [comment]
I no longer think this is a good idea. We chose a different approach where we communicate with all nodes before the start of the protocol, ensuring a high success rate.
If anybody thinks this is still relevant, I'm happy to discuss this.
